internal/common/crypto_policy.go

```diff
@@ -0,0 +1,126 @@
+package common
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	ocpv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/crypto"
+	"kubevirt.io/ssp-operator/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type SSPTLSOptions struct {
+	MinTLSVersion      string
+	OpenSSLCipherNames []string
+}
+
+func (s *SSPTLSOptions) IsEmpty() bool {
+	return len(s.OpenSSLCipherNames) == 0 && s.MinTLSVersion == ""
+}
+
+func (s *SSPTLSOptions) MinTLSVersionId() (minTlsVersionId uint16) {
+	switch s.MinTLSVersion {
+	case "1.0":
+		return tls.VersionTLS10
+	case "1.1":
+		return tls.VersionTLS11
+	case "1.2":
+		return tls.VersionTLS12
+	case "1.3":
+		return tls.VersionTLS13
+	default:
+		return 0
+	}
+}
+
+func NewSSPTLSOptions(tlsSecurityProfile *ocpv1.TLSSecurityProfile, logger *logr.Logger) (options *SSPTLSOptions) {
+	ciphers, tlsProfile := selectCipherSuitesAndMinTLSVersion(tlsSecurityProfile)
+
+	if logger != nil {
+		logger.Info("Got Ciphers and tlsProfile:", "ciphers: ", ciphers, "tlsProfile: ", tlsProfile)
+	}
+
+	return &SSPTLSOptions{
+		MinTLSVersion:      tlsVersionToHumanReadable(tlsProfile),
+		OpenSSLCipherNames: ciphers,
+	}
+}
+
+func (s *SSPTLSOptions) CipherIDs() (cipherSuites []uint16) {
+	for _, cipherName := range crypto.OpenSSLToIANACipherSuites(s.OpenSSLCipherNames) {
+		for _, knownCipher := range tls.CipherSuites() {
+			if knownCipher.Name == cipherName {
+				cipherSuites = append(cipherSuites, knownCipher.ID)
+			}
+		}
+	}
+	return
+}
+
+func emptyTLSProfile() *SSPTLSOptions {
+	return &SSPTLSOptions{"", []string{}}
+}
+
+var (
+	setupLog = ctrl.Log.WithName("setup")
```

Comment:
If `setupLog` is only used in `GetCiperSuitesAndMinTLSVersion` we should declare it just in this function.

Reply:
I like to keep the log separate cause the next developer adding a log in some other method will find it easy, even if we use it only there currently, does it make sense? if you find it is not worth it - I will move it.

Reply:
I would stick to declaring it only in the function because this makes it a "global" variable to the whole `common` package and it is not used anywhere else.